### PR TITLE
Restore writeups for GpioTestTool, I2cTestTool, SpiTestTool

### PR DESCRIPTION
**Pull request by jordanrh1**

### CODE REVIEW

File: en-US/Samples/GpioTestTool.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+layout: sample
+title: GpioTestTool
+permalink: /en-US/Samples/GpioTestTool.htm
+lang: en-US
+---
+
+# GpioTestTool Sample
```
Comment (stephenhuang) [inline en-US/Samples/GpioTestTool.md]: change to # GpioTestTool

File: en-US/Samples/GpioTestTool.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+---
+layout: sample
```
Comment (stephenhuang) [inline en-US/Samples/GpioTestTool.md:2]: Please update the header to follow the new template
https://github.com/ms-iot/content/blob/develop/Resources/contribute/template/sample-template.md

Including a link to the gpiotesttool on GitHub.com/samples.  (i'll make sure that they go live at the exact same time)

The link would be www.github.com/ms-iot/samples/gpiotesttool

File: en-US/Samples/GpioTestTool.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+GpioTestTool is a simple utility that allows you to write and
+read GPIO pins on the command line. GpioTestTool is written in standard
+C++ and consumes the Windows.Devices.Gpio WinRT APIs at the ABI level
+with the help of the
+[Windows Runtime Library (WRL)](https://msdn.microsoft.com/en-us/library/hh438466.aspx).
+These techniques can be used to consume most WinRT APIs from native
+applications.
+
+### Usage
```
Comment (stephenhuang) [inline en-US/Samples/GpioTestTool.md]: ## Usage

---

File: en-US/Samples/GpioTestTool.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      Debounce Timeout: 0
+            Drive Mode: input
+    > read
+    High
+    > setdrivemode output
+    > write 0
+    > write 1
+    > q
+
+### Building and running the sample
```
Comment (stephenhuang) [inline en-US/Samples/GpioTestTool.md]: ## Building and running the sample

---

File: en-US/Samples/GpioTestTool.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+1. Open `GpioTestTool\GpioTestTool.sln` in Visual Studio.
+1. Select the target architecture.
+   - Select `ARM` for Raspberry Pi or DragonBoard 410c
+   - Select `x86` for MinnowBoardMax
+1. Go to `Build -> Build Solution`
+1. Copy GpioTestTool.exe from the build output folder to your device.
+1. SSH into your device and run `GpioTestTool.exe` (with the appropriate
+   command line parameters, of course).
+
+### The code
```
Comment (stephenhuang) [inline en-US/Samples/GpioTestTool.md]: ## The code

---

File: en-US/Samples/I2cTestTool.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
+---
+layout: sample
```
Comment (stephenhuang) [inline en-US/Samples/I2cTestTool.md:2]: As mentioned above, please follow the new header

File: en-US/Samples/I2cTestTool.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+title: I2cTestTool
+permalink: /en-US/Samples/I2cTestTool.htm
+lang: en-US
+---
+
+# I2cTestTool Sample
```
Comment (stephenhuang) [inline en-US/Samples/I2cTestTool.md]: change to # I2cTestTool

File: en-US/Samples/I2cTestTool.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+
+
+[View the code on Github](https://github.com/ms-iot/samples/blob/develop/I2cTestTool/main.cpp)
+
+I2cTestTool is a utility for interacting with I2C devices on the command
+line. I2cTestTool is written in C++/CX and shows how to consume WinRT components
+from command line applications. The resulting tool is a useful debugging aid.
+
+### Usage
```
Comment (stephenhuang) [inline en-US/Samples/I2cTestTool.md]: ## Usage

---

File: en-US/Samples/I2cTestTool.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+            Bus Speed: StandardMode (100Khz)
+      > write {0 1 2 3 4 5 6 7 8}
+      > writeread {4} 4
+       5 6 7 8
+      > write {0}
+      > read 8
+       1 2 3 4 5 6 7 8
+      > q
+
+### Building and running the sample
```
Comment (stephenhuang) [inline en-US/Samples/I2cTestTool.md]: ## Building and running the sample

---

File: en-US/Samples/I2cTestTool.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+1. Open `I2cTestTool\I2cTestTool.sln` in Visual Studio
+1. Select the target architecture.
+   - Select `ARM` for Raspberry Pi
+   - Select `x86` for MinnowBoardMax
+1. Go to `Build -> Build Solution`
+1. Copy `I2cTestTool.exe` from the build output folder to your device.
+1. SSH into your device and run `I2cTestTool.exe` (with required command
+   line parameters, of course)
+
+### Creating a blank C++/CX console application
```
Comment (stephenhuang) [inline en-US/Samples/I2cTestTool.md]: ## Creating a blank C++/CX console application

---

File: en-US/Samples/SpiTestTool.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
+---
+layout: sample
```
Comment (stephenhuang) [inline en-US/Samples/SpiTestTool.md:2]: Please update header

File: en-US/Samples/SpiTestTool.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+title: SpiTestTool
+permalink: /en-US/Samples/SpiTestTool.htm
+lang: en-US
+---
+
+# SpiTestTool Sample
```
Comment (stephenhuang) [inline en-US/Samples/SpiTestTool.md]: # SpiTestTool

File: en-US/Samples/SpiTestTool.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+
+
+[View the code on Github](https://github.com/ms-iot/samples/blob/develop/SpiTestTool/main.cpp)
+
+SpiTestTool is a utility for interacting with SPI devices on the command
+line. SpiTestTool is written in C++/CX and shows how to consume WinRT components
+from command line applications. The resulting tool is a useful debugging aid.
+
+### Usage
```
Comment (stephenhuang) [inline en-US/Samples/SpiTestTool.md]: ## Usage

---

File: en-US/Samples/SpiTestTool.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+      > write {1 2 3 4}
+      > read 8
+       0 ff ff ff ff ff ff ff
+      > writeread {aa bb} 4
+       bb ff ff ff
+      > fullduplex {aa bb cc dd}
+       0 aa b7 99
+      > q
+
+### Building and running the sample
```
Comment (stephenhuang) [inline en-US/Samples/SpiTestTool.md]: ## Building and running the sample

---

**Review by stephenhuang — COMMENTED** [review]
We've made some minor styling updates to our samples. Please see below

**Comment by stephenhuang** [comment]
In addition, please update the samples.json file 
instructions here: https://github.com/ms-iot/content/blob/develop/Resources/contribute/authoring-guidelines.md

**Comment by jordanrh1** [comment]
Thanks for the review, I have updated the PR.

**Pull request merged**